docs/manuals/gui/extension/dynamic_element/index.md

```diff
@@ -0,0 +1,358 @@
+the type it supports. This is handled by the `PropertyType^` class.<br/>
+Every custom element that wants to use a property with a *dynamic* type
+must be defined as a *dynamic element*.<br/>
+Although a custom element might not have any *dynamic property* (that is, a property
+with a type that is dynamic), it can be implemented as a *dynamic element* nevertheless
+to leverage the expressivity of how these elements are implemented.
+
+Dynamic elements use [TypeScript](https://www.typescriptlang.org/) and
+[JavaScript](https://www.javascript.com/) code to dynamically generate HTML code to
+produce the pages that can be displayed in a browser. Taipy GUI relies
+on the [React](https://reactjs.org/) JavaScript library to simplify the
+development of graphical components.
+
+!!! warning "Minimal knowledge"
+    The following sections contain code samples written in TypeScript that leverage
+    the React library. This manual is in no way a beginner's guide on either technology.
+
+    To create your custom dynamic elements, you need to know about React functional
+    components and hooks and how to create simple components.<br/>
+    You also need to know the basics about packaging React components using
+    [webpack](https://webpack.js.org/) that bundles the JavaScript modules into
+    a JavaScript library that element libraries can load to implement the front-end
+    part of the custom elements.
+
+## Declaring a dynamic element
+
+The declaration of a dynamic element looks very similar to the declaration of a static
+element.<br/>
+The fundamental change is that if the *render_xhtml* argument of the
+[`Element` constructor](Element.__init__()^) is not set or is not a function
+then the element is considered *dynamic*. That is, implemented using a React
+component. You can specify the name of the component using the *react_component*
+argument. If you don't, Taipy GUI will use a capitalized camel case transformation
+of the element name as the target React component name.
+
+Here is how a custom element library containing a dynamic element would be declared:
+
+```py
+from taipy.gui.extension import ElementLibrary, Element, ElementProperty
+
+class CustomLibrary(ElementLibrary):
+
+    def __init__(self) -> None:
+        self.elements = {
+              ...
+              "<element1 name>": Element("<default property name>", {
+                  "<property1 name>": ElementProperty(<property1 type>, ...),
+                  ...
+              },
+              # This is optional and can be omitted if the component name can be inferred
+              # from the element name
+              react_component="<element1 component name>"),
+              ...
+        }
+
+    def get_name(self) -> str:
+        return "<library name>"
+
+    def get_elements(self) -> dict:
+        return self.elements
+
+    def get_scripts(self) -> list[str]:
+        return ["<javascript module pathname>"]
```

Comment:
Pathname or path?

Reply:
I kind of associate 'path' to where a directory is located, and 'pathname' to the full path to a file, including the file name...